src/pages/trade/ui/order-form/order-form-store.ts

```diff
@@ -0,0 +1,375 @@
+
+    this.baseAsset.onAmountChange(
+      debouncedHandleAmountChange as (asset: OrderFormAsset) => Promise<void>,
+    );
+    this.quoteAsset.onAmountChange(
+      debouncedHandleAmountChange as (asset: OrderFormAsset) => Promise<void>,
+    );
+
+    this.setBalancesOfAssets();
+    void this.calculateGasFee();
+    void this.calculateExchangeRate();
+  };
+
+  setBalances = (balances: BalancesResponse[]): void => {
+    this.balances = balances;
+    this.setBalancesOfAssets();
+  };
+
+  handleAmountChange = async (asset: OrderFormAsset): Promise<void> => {
+    const assetIsBaseAsset = asset.assetId === this.baseAsset.assetId;
+    const assetIn = assetIsBaseAsset ? this.baseAsset : this.quoteAsset;
+    const assetOut = assetIsBaseAsset ? this.quoteAsset : this.baseAsset;
+
+    try {
+      void this.calculateGasFee();
+
+      assetOut.setIsEstimating(true);
+
+      // reset potentially previously set flag
+      assetIn.setIsApproximately(false);
+
+      const output = await this.simulateSwapTx(assetIn, assetOut);
+      if (!output) {
+        return;
+      }
+
+      const outputAmount = getFormattedAmtFromValueView(output, true);
+
+      assetOut.setAmount(Number(outputAmount), false);
+      assetOut.setIsApproximately(true);
+    } finally {
+      assetOut.setIsEstimating(false);
+    }
+  };
+
+  calculateExchangeRate = async (): Promise<void> => {
+    this.exchangeRate = null;
+
+    const baseAsset: OrderFormAsset = new OrderFormAsset(this.baseAsset.metadata);
+    baseAsset.setAmount(1);
+
+    const output = await this.simulateSwapTx(baseAsset, this.quoteAsset);
+    if (!output) {
+      return;
+    }
+
+    const outputAmount = getFormattedAmtFromValueView(output, true);
+    this.exchangeRate = Number(outputAmount);
+  };
+
+  calculateGasFee = async (): Promise<void> => {
+    this.gasFee = null;
+
+    const isBuy = this.direction === Direction.Buy;
+    const assetIn = isBuy ? this.quoteAsset : this.baseAsset;
+    const assetOut = isBuy ? this.baseAsset : this.quoteAsset;
+
+    if (!assetIn.amount || !assetOut.amount) {
+      this.gasFee = 0;
+      return;
+    }
+
+    const req = new TransactionPlannerRequest({
+      swaps: [
+        {
+          targetAsset: assetOut.assetId,
+          value: {
+            amount: assetIn.toAmount(),
+            assetId: assetIn.assetId,
+          },
+          claimAddress: assetIn.accountAddress,
+        },
+      ],
+      source: assetIn.accountIndex,
+    });
+
+    const txPlan = await plan(req);
+    const fee = txPlan.transactionParameters?.fee;
+    const feeValueView = new ValueView({
+      valueView: {
+        case: 'knownAssetId',
+        value: {
+          amount: fee?.amount ?? { hi: 0n, lo: 0n },
+          metadata: this.baseAsset.metadata,
+        },
+      },
+    });
+
+    const feeAmount = getFormattedAmtFromValueView(feeValueView, true);
+    this.gasFee = Number(feeAmount);
+  };
+
+  initiateSwapTx = async (): Promise<void> => {
```

Comment:
take note of the differences in the toast notifications between these swap transactions. The dex-explorer variant has some miscoloration and unnecessary checkboxes – we should try to emulate the former.

https://github.com/user-attachments/assets/5dde3e9a-7e35-4695-a9c3-c189991fbb96




https://github.com/user-attachments/assets/b87c7266-cd1b-4709-bc49-2f6456509ab3


<img width="343" alt="Screenshot 2024-11-19 at 9 57 04 PM" src="https://github.com/user-attachments/assets/826a1f02-8637-4a98-8e88-37d5f6dea3bb">
<img width="343" alt="Screenshot 2024-11-19 at 9 57 20 PM" src="https://github.com/user-attachments/assets/9d7c10e7-b682-40fc-8aad-7c9b180bfd0b">

Reply:
Good find, will address this.